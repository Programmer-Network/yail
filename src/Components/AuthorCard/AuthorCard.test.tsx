import { fireEvent, render, screen } from "@testing-library/react";
import { vi } from "vitest";

import { AuthorCard } from ".";

describe("AuthorCard component", () => {
  const mockProps = {
    name: "John Doe",
    avatar: "https://avatars.githubusercontent.com/u/2085263?v=4",
    about: "An expert in the field."
  };

  it("renders correctly", () => {
    const mockProps = {
      name: "John Doe",
      avatar: "https://avatars.githubusercontent.com/u/2085263?v=4",
      about: "An expert in the field."
    };

    const { asFragment } = render(<AuthorCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders author's name, avatar, and about text", () => {
    render(<AuthorCard {...mockProps} />);

    expect(screen.getByText(mockProps.name)).toBeInTheDocument();
    expect(screen.getByText(mockProps.about)).toBeInTheDocument();
    expect(screen.getByRole("img", { name: mockProps.name })).toHaveAttribute(
      "src",
      mockProps.avatar
    );
  });

  test("calls onClick with author details when the card is clicked", () => {
    const handleClick = vi.fn();
    render(<AuthorCard {...mockProps} onClick={handleClick} />);

    fireEvent.click(screen.getByText(mockProps.name));
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(handleClick).toHaveBeenCalledWith({
      name: mockProps.name,
      avatar: mockProps.avatar,
      about: mockProps.about
    });
  });
});
