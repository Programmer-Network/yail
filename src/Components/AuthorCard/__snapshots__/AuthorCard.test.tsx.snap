// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthorCard component > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="border-border rounded-lg border-2 p-6"
  >
    <div
      class="flex flex-col items-center gap-4 md:flex-row"
    >
      <div
        class="hidden md:block"
      >
        <div
          class="sm:mt-0 rounded-full overflow-hidden flex items-center justify-center flex-shrink-0"
          style="width: 75px; height: 75px;"
        >
          <img
            alt="John Doe"
            class="bg-background h-full w-full object-cover object-center"
            loading="lazy"
            src="https://avatars.githubusercontent.com/u/2085263?v=4"
          />
        </div>
      </div>
      <div>
        <h3
          class="text-xl sm:text-2xl md:text-3xl text-text break-words font-bold"
        >
          John Doe
        </h3>
        <p
          class="text-text break-words overflow-hidden"
        >
          An expert in the field.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
